velox/functions/prestosql/aggregates/tests/AverageAggregationTest.cpp

```diff
@@ -534,5 +534,55 @@ TEST_F(AverageAggregationTest, constantVectorOverflow) {
   assertQuery(plan, "SELECT 1073741824");
 }
 
+TEST_F(AverageAggregationTest, companionFunctionsWithNonFlatInputs) {
```

Comment:
should we check for the case where either one of children is flat? can there be multiple encoding layers?